commit dcac53db5e727521fe0a13c1a56505ac79255728
Author: Emmanuele Bassi <ebassi@gnome.org>
Date:   Sat Mar 27 12:32:04 2021 +0100

    Do not run local-dependent tests with missing locales
    
    Not everyone has all the locales installed all the time, so let's check
    if a locale is available before using it.

diff --git a/libgweather/test_libgweather.c b/libgweather/test_libgweather.c
index b10c1f16..f97683a2 100644
--- a/libgweather/test_libgweather.c
+++ b/libgweather/test_libgweather.c
@@ -678,6 +678,7 @@ test_location_names (void)
 {
     g_autoptr(GWeatherLocation) world = NULL;
     g_autoptr(GWeatherLocation) brussels = NULL;
+    char *old_locale;
 
     world = gweather_location_get_world ();
     g_assert_nonnull (world);
@@ -689,7 +690,15 @@ test_location_names (void)
     g_assert_cmpstr (gweather_location_get_english_name (brussels), ==, "Brussels");
     gweather_location_unref (brussels);
 
+    old_locale = g_strdup (setlocale (LC_ALL, NULL));
     setlocale (LC_ALL, "fr_FR.UTF-8");
+    if (strstr (setlocale (LC_ALL, NULL), "fr_FR.UTF-8") == NULL)
+      {
+        g_test_skip ("locale fr_FR.UTF-8 not available, skipping localization tests");
+        setlocale (LC_ALL, old_locale);
+        g_free (old_locale);
+        return;
+      }
 
     g_clear_pointer (&world, gweather_location_unref);
     g_clear_pointer (&brussels, gweather_location_unref);
@@ -706,6 +715,7 @@ test_location_names (void)
     gweather_location_unref (brussels);
 
     setlocale (LC_ALL, "");
+    setlocale (LC_ALL, old_locale);
     g_clear_pointer (&world, gweather_location_unref);
     g_clear_pointer (&brussels, gweather_location_unref);
     _gweather_location_reset_world ();
